### PR TITLE
Added generic upgrade procedure

### PR DESCRIPTION
**Pull request by rkothiya**
Signed-off-by: Rinku Kothiya <rkothiya@redhat.com>

### CODE REVIEW

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+### Online upgrade procedure for servers
+This procedure involves upgrading **one server at a time**, while keeping the volume(s) online and client IO ongoing. This procedure assumes that multiple replicas of a replica set, are not part of the same server in the trusted storage pool.
+
+> **ALERT:** If there are disperse or, pure distributed volumes in the storage pool being upgraded, this procedure is NOT recommended, use the [Offline upgrade procedure](#offline-upgrade-procedure) instead.
+
+#### Repeat the following steps, on each server in the trusted storage pool, to upgrade the entire pool to new-version :
+1. Stop all gluster services, either using the command below, or through other means,
+
+        # killall glusterfs glusterfsd glusterd
```
Comment (kshithijiyer) [inline docs/Upgrade-Guide/Generic_Upgrade_procedure.md:20]: Ideally glusterd should be stopped using systemctl.  so it should be
```
# systemctl stop glusterd
# killall glusterfs glusterfsd
```

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+3. Install Gluster new-version
+
+4. Ensure that version reflects new-version in the output of,
+
+        # gluster --version
+
+5. Start glusterd on the upgraded server
+
+        # glusterd
```
Comment (kshithijiyer) [inline docs/Upgrade-Guide/Generic_Upgrade_procedure.md:33]: This should be `systemctl start glusterd` as starting glusterd using `glusterd` command create a bunch of issues.

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+1. Stop all gluster services, either using the command below, or through other means,
+
+        # killall glusterfs glusterfsd glusterd
+        # systemctl stop glustereventsd
+
+2. Stop all applications that run on this server and access the volumes via gfapi (qemu, NFS-Ganesha, Samba, etc.)
+
+3. Install Gluster new-version
+
+4. Ensure that version reflects new-version in the output of,
```
Comment (kshithijiyer) [inline docs/Upgrade-Guide/Generic_Upgrade_procedure.md:27]: `# gluster --version` command will not run if glusterd is down, you have to move this point after point 5. So ideally point 5 would be point 4 and point 4 would be point 5.

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+10. Restart any gfapi based application stopped previously in step (2)
+
+### Offline upgrade procedure
+This procedure involves cluster downtime and during the upgrade window, clients are not allowed access to the volumes.
+
+#### Steps to perform an offline upgrade:
+1. On every server in the trusted storage pool, stop all gluster services, either using the command below, or through other means,
+
+```sh
+    #killall glusterfs glusterfsd glusterd glustereventsd
```
Comment (kshithijiyer) [inline docs/Upgrade-Guide/Generic_Upgrade_procedure.md:62]: Same as above also add space after #.

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    #systemctl stop glustereventsd
```
Comment (kshithijiyer) [inline docs/Upgrade-Guide/Generic_Upgrade_procedure.md:63]: Add space after #.

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+```
+2. Stop all applications that access the volumes via gfapi (qemu, NFS-Ganesha, Samba, etc.), across all servers
+
+3. Install Gluster new-version, on all servers
+
+4. Ensure that version reflects new-version in the output of the following command on all servers,
+```sh
+    #gluster --version
```
Comment (kshithijiyer) [inline docs/Upgrade-Guide/Generic_Upgrade_procedure.md:71]: Add a space after #

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+```
+
+5. Start glusterd on all the upgraded servers
+```sh
+    #glusterd
```
Comment (kshithijiyer) [inline docs/Upgrade-Guide/Generic_Upgrade_procedure.md:76]: Same as above

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+```
+6. Ensure that all gluster processes are online by checking the output of,
+```sh
+    #gluster volume status
```
Comment (kshithijiyer) [inline docs/Upgrade-Guide/Generic_Upgrade_procedure.md:80]: Add space after #

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+```
+
+7. If the glustereventsd service was previously enabled, it is required to start it using the commands below, or, through other means,
+```sh
+    #systemctl start glustereventsd
```
Comment (kshithijiyer) [inline docs/Upgrade-Guide/Generic_Upgrade_procedure.md:85]: Add space after #

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md (hunk 1 above)
Comment (rkothiya) [reply docs/Upgrade-Guide/Generic_Upgrade_procedure.md:20]: Done, as discussed will also keep 
# systemctl stop glustereventsd

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md (hunk 2 above)
Comment (rkothiya) [reply docs/Upgrade-Guide/Generic_Upgrade_procedure.md:33]: Done

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md (hunk 3 above)
Comment (rkothiya) [reply docs/Upgrade-Guide/Generic_Upgrade_procedure.md:27]: I tried this once more after reading you comment and even though glusterd was not running we can run gluster --version command. Hence keeping this as it is.

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md (hunk 4 above)
Comment (rkothiya) [reply docs/Upgrade-Guide/Generic_Upgrade_procedure.md:62]: Done

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md (hunk 5 above)
Comment (rkothiya) [reply docs/Upgrade-Guide/Generic_Upgrade_procedure.md:63]: Done

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md (hunk 6 above)
Comment (rkothiya) [reply docs/Upgrade-Guide/Generic_Upgrade_procedure.md:71]: Done

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md (hunk 8 above)
Comment (rkothiya) [reply docs/Upgrade-Guide/Generic_Upgrade_procedure.md:80]: Done

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md (hunk 9 above)
Comment (rkothiya) [reply docs/Upgrade-Guide/Generic_Upgrade_procedure.md:85]: Done

File: docs/Upgrade-Guide/Generic_Upgrade_procedure.md (hunk 3 above)
Comment (kshithijiyer) [reply docs/Upgrade-Guide/Generic_Upgrade_procedure.md:29]: Thanks for letting me know :) something new for me as well

**Pull request merged**